RizinCpp.h:110:34: warning: 'iterator<std::input_iterator_tag, char *>' is deprecated

**Environment information**
* Operating System: Mac OS 12.5.1 on ARM64
* Cutter version: https://github.com/rizinorg/cutter/commit/e6db27135dcb9f2aa46bde91ac5c82e64ed51df3
* Obtained from:
  - [x] Built from source
  - [ ] Downloaded release from Cutter website or GitHub 
  - [ ] Distribution repository
* File format: -

**Describe the bug**

```
/Users/user/rizin/cutter/src/core/RizinCpp.h:110:34: warning: 'iterator<std::input_iterator_tag, char *>' is deprecated [-Wdeprecated-declarations]
    class iterator : public std::iterator<std::input_iterator_tag, T *>
                                 ^
/Users/user/rizin/cutter/src/widgets/Dashboard.cpp:141:30: note: in instantiation of member class 'CutterRzList<char>::iterator' requested here
        for (const auto &lib : CutterRzList<char>(libs)) {
                             ^
/Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/__iterator/iterator.h:27:29: note: 'iterator<std::input_iterator_tag, char *>' has been explicitly marked deprecated here
struct _LIBCPP_TEMPLATE_VIS _LIBCPP_DEPRECATED_IN_CXX17 iterator
                            ^
/Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/__config:1066:39: note: expanded from macro '_LIBCPP_DEPRECATED_IN_CXX17'
#  define _LIBCPP_DEPRECATED_IN_CXX17 _LIBCPP_DEPRECATED
                                      ^
/Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/__config:1043:48: note: expanded from macro '_LIBCPP_DEPRECATED'
#    define _LIBCPP_DEPRECATED __attribute__ ((deprecated))
                                               ^
1 warning generated.
```

--- a/src/core/RizinCpp.h
+++ b/src/core/RizinCpp.h
@@ -71,8 +71,15 @@ private:
     const RzPVector *const vec;
 
 public:
-    class iterator : public std::iterator<std::input_iterator_tag, T *>
+    class iterator
     {
+    public:
+        using iterator_category = std::input_iterator_tag;
+        using value_type = T *;
+        using difference_type = ptrdiff_t;
+        using pointer = T **;
+        using reference = T *&;
+
     private:
         T **p;
 
@@ -107,8 +114,15 @@ private:
     const RzList *const list;
 
 public:
-    class iterator : public std::iterator<std::input_iterator_tag, T *>
+    class iterator
     {
+    public:
+        using iterator_category = std::input_iterator_tag;
+        using value_type = T *;
+        using difference_type = ptrdiff_t;
+        using pointer = T **;
+        using reference = T *&;
+
     private:
         RzListIter *iter;
 

